packages/amplify-schema-validator/src/helpers/resolve-field-type-name.ts

```diff
@@ -0,0 +1,30 @@
+import { NamedTypeNode, TypeNode } from 'graphql';
+
+export const resolveFieldTypeName = (type: TypeNode | undefined): string => {
+  switch (type?.kind) {
+    case 'NamedType': {
+      return type.name.value;
+    }
+    case 'NonNullType':
+      if (type.type.kind === 'NamedType') {
+        return type.type.name.value;
+      } if (type.type.kind === 'ListType') {
+        if (type.type.type.kind === 'NamedType') {
+          return type.type.type.name.value;
+        } if (type.type.type.kind === 'NonNullType') {
+          return (type.type.type.type as NamedTypeNode).name.value;
+        } return '';
+      } return '';
+
+    case 'ListType': {
+      if (type.type.kind === 'NamedType') {
+        return type.type.name.value;
+      } if (type.type.kind === 'NonNullType') {
+        return (type.type.type as NamedTypeNode).name.value;
+      } return '';
+    }
+    default: {
+      return '';
+    }
```

Comment:
Is a string a reasonable sentinel value here, or should we return an undefined here?

Reply:
I prefer undefined - it's more explicit. updated!